### ISSUE TITLE
Google Plus Login issues

### ISSUE THREAD

**Issue by PGMacDesign** [CloudRail/cloudrail-si-android-sdk#14]
Howdy all, 
     working with the Google Plus login today and encountered an error. 
     My setup for the Google Plus OAuth was as follows (at the bottom): 

     It ran no problem and got past the 2 step authentication. After that screen however, it changed over to this image:  http://screencloud.net/v/4fye

     It looks like since I am unable to change said redirect URI, it might need to be changed from "https://www.cloudrailauth.com/auth " to the generic localhost, not sure. 

     Either way, the code is located in the GooglePlus.class on line 63.  

//Code
final Profile profile = new GooglePlus(context, GOOGLE_CLIENT_ID, GOOGLE_CLIENT_SECRET);

**Comment by FlorianWendel** [comment]
Hi there, 
can you check if it works when you follow these instructions: https://docs.cloudrail.com/docs/how-to-get-the-credentials#google-plus

Correctly configuring an app with Google is a bit tricky , since there are lots of options.

**Comment by limelect** [comment]
I will tomorrow morning and report


On 21/06/2016 23:58, Florian Wendel wrote:
>
> Hi there,
> can you check if it works when you follow these instructions: 
> https://docs.cloudrail.com/docs/how-to-get-the-credentials#google-plus
>
> Correctly configuring an app with Google is a bit tricky , since there 
> are lots of options.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/CloudRail/cloudrail-si-android-sdk/issues/14#issuecomment-227569747>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AEAS24-nzh3l6PNRA10f0mv-EXb5syvNks5qOFCJgaJpZM4I7J8C>.
>

-- 

Yours


Shlomo Abuisak 0505-444457

Lim Electronics P.O.B 1149 Givatayim

Israel


limelect@gmail.com

_*http://www.limelect.com*_ <http://www.limelect.com/>

**Comment by limelect** [comment]
Sorry to misunderstood

At a time i opened API keys and got the ID's and secret key's.

I do not wants to use your app as is. I am interested

only on the basic idea. And develop my own.

So i am trying to understand how can i help ?


On 21/06/2016 23:58, Florian Wendel wrote:
>
> Hi there,
> can you check if it works when you follow these instructions: 
> https://docs.cloudrail.com/docs/how-to-get-the-credentials#google-plus
>
> Correctly configuring an app with Google is a bit tricky , since there 
> are lots of options.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/CloudRail/cloudrail-si-android-sdk/issues/14#issuecomment-227569747>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AEAS24-nzh3l6PNRA10f0mv-EXb5syvNks5qOFCJgaJpZM4I7J8C>.
>

-- 

Yours


Shlomo Abuisak 0505-444457

Lim Electronics P.O.B 1149 Givatayim

Israel


limelect@gmail.com

_*http://www.limelect.com*_ <http://www.limelect.com/>

**Comment by PGMacDesign** [comment]
Ok, got it cleaned up. so, Florian, you were correct on both issues.
**First**, I had to switch it from Android credentials to Web and second, **Second**, the redirect URI needed to be whitelisted on the webpage itself. Once those two were in place, I was able to get it working. 

Here is the visual if anyone else gets this error: http://screencloud.net/v/mD3K

On an unrelated note, I just noticed that the returned String from profile.saveAsString() is different for the various OAuth logins. Twitter and Google for example have a different structure to them where twitter uses the Strings "oauthToken", and "oauthTokenSecret", Google uses "refreshToken" and "accessToken". By chance, are these listed within the documentation at all? I freely admit I did not read them in their entirety and may have missed them. If not, is it possible to be added in the future?
